python/pydantic_core/core_schema.py

```diff
@@ -2857,6 +2857,7 @@ class TypedDictSchema(TypedDict, total=False):
     type: Required[Literal['typed-dict']]
     fields: Required[Dict[str, TypedDictField]]
     cls: Type[TypedDict]
+    generic_origin: Type[TypedDict]
```

Comment:
If we do that, presumably we'd want to make that change in `cls` above as well, since it's functioning equivalently. I think if it's not raising errors or otherwise misbehaving though it's not a bad idea to keep as-is since it makes sense (at least I think it does). If it's type-checking incorrectly though I don't mind changing.

Reply:
My comment was indeed on the two lines :)

It makes sense conceptually yes although type checkers won't understand this (i.e. the won't complain when passing something else than a typeddict class and it won't understand access to the class attributes (`__required_keys__`, etc)).

It may still pass type checking because we are on an older pyright version on pydantic-core that did not emit any error at that time.

Reply:
Also, I mentioned this on slack but just making the point again here — it seems that this field isn't currently used in this PR, and more generally I'm not sure it's even possible to use it since we have no runtime instance information that could even be used to short-circuit the full revalidation anyway. So I'm thinking we probably already get the desired behavior for generic TypedDicts (though note that that is just because we presumably _always_ do a full revalidation, a downside of passing TypedDict subclass instances through multiple pydantic validators that doesn't exist for BaseModel or dataclasses)

Reply:
Yeah, will remove the typed dict stuff :) I don't think it's actually necessary, I realized when hooking things up to the `pydantic` end of things.

Reply:
Going to resolve for now. Happy to change that type hint in another PR for `cls` if you think that's appropriate cc @Viicos.